### PR TITLE
Some style tweaks

### PR DESCRIPTION
**Pull request by Benestar**
Removed some redundant methods and added some comments.

### CODE REVIEW

File: src/Deserializers/LegacySnakDeserializer.php [hunk 1]
```diff
@@ -41,11 +42,7 @@ public function deserialize( $serialization ) {
 				return new PropertySomeValueSnak( $serialization[1] );
 			case 'value':
 				return $this->deserializeValueSnak( $serialization );
-				// @codeCoverageIgnoreStart
-			default:
-				throw new LogicException();
```
Comment (thiemowmde) [inline src/Deserializers/LegacySnakDeserializer.php]: -1. The code should fail if that happens and not silently return null.

**Comment by thiemowmde** [comment]
+1 to the inlining. -1 to the removed exception.

File: src/Deserializers/LegacySnakDeserializer.php (hunk 1 above)
Comment (Benestar) [inline src/Deserializers/LegacySnakDeserializer.php]: Note that this cannot happen. It is impossible that the code reaches this line of code. We should not harvest dead code.

Comment (thiemowmde) [inline src/Deserializers/LegacySnakDeserializer.php]: This is not just "dead code". It's there for a purpose. PHPStorm and other tools complain about this for a reason. The fact that you, when following all execution paths, are able to understand that no code path in the current implementation can reach this line does not mean all other people reading the code will understand this. More important: It does not mean these code paths will never change and never reach this line.

So still -1.

**Comment by JeroenDeDauw** [comment]
I'm not a fan of removing these methods. The methods in which they where used are now slightly harder to read due to pollution by things from a too low level of abstraction.

**Comment by Benestar** [comment]
Removed all controversial stuff...

**Pull request merged**